src/Dfc.CourseDirectory.WebV2/Features/Providers/EditProviderTypeController.cs

```diff
@@ -30,11 +30,18 @@ public async Task<IActionResult> Get(ProviderContext providerContext)
             EditProviderType.Command command,
             ProviderContext providerContext)
         {
-            command.ProviderId = providerContext.ProviderInfo.ProviderId;
+            if (command.ProviderId != providerContext.ProviderInfo.ProviderId)
+            {
+                return BadRequest();
+            }
+
```

Comment:
```suggestion
            command.ProviderId = providerContext.ProviderInfo.ProviderId;
```

Reply:
As mentioned earlier, this is deliberate because provider context can be determined by session at the moment. We can remove it once this is no longer the case.